CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,76 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as
+contributors and maintainers pledge to making participation in our project and
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, sex characteristics, gender identity and expression,
+level of experience, education, socio-economic status, nationality, personal
+appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+ advances
```

Comment:
The inclusion of the last bit about _unwelcome_ sexual attention or advances can be problematic, despite the good intentions that motivate it.

Sexual relationships, by nature, are about individuals and vulnerability. There are lots of gray areas, no universally established rules. Therefore, sexual relationships are **not** safe. Situations can become quite messy. Even when two (or more?) people are in a consensual relationship, there is no way to know for sure, in advance, when sexual attention or advances will be welcomed. Even when those advances are made in good faith, with the intention to connect with the other person, not to harass or coerce them to do something that they don't want. Or in the context of a history of previously accepted advances.

Furthermore, the reaction of person A to what person B did does not determine the rightness or wrongness of person's B action. For example, think about a parent punishing his/her children because of their poor performance at school by prohibiting them from playing video-games for a week. The children may not like it, they may react negativelly. That does not mean that the parent is doing something wrong. Likewise, just because a sexual advance/attention is unwelcomed, that doesn't mean it's a crime, or that it was done with bad intentions. It could be the case of the person having underdeveloped social skills or missreading the situation - remember the previous point, there's no way to know for sure in advance. Should the person be punished just for that?

I recognize the importance of fostering an environment where contributors can interact with each other in a civil and constructive manner. What bothers me about this part of the code of conduct is the lack of nuance in approaching a complex issue. This is not something particular to this document, but a reflection of the overall shallowness with which difficult issues such as these are being discussed in our culture, often taking a reductionist approach where "consent" is seen as a simple solution to turn sexual relationships into safe interactions, which they are not. For a further discussion on this, see: [Consent isn't everything and sex is not like tea](https://quillette.com/2018/10/19/consent-isnt-everything-and-sex-is-not-like-tea/).

My opinion is that project maintainers, in general, should not meddle in personal interactions between contributors. I recommend removing these lines, or rewriting them, taking into account what's presented here (though I confess not even I know how to do that right now). We should, however, make a note to always remind them to be respectful to each other and stick to the code and relevant issues when commenting/reviewing. When problems appear, we should listen to all involved parties, consider not only reactions but also intentions, respect people's privacy and reflect if we are indeed the most adequate channel to judge and treat the issue presented.

Sometimes it will be just a matter of sitting down, talking and clarifying things. Maybe in some occasion we will have to redirect it to HR. Hopefully we will **never** have to get in touch with the police, though we should not discard this possibility if the situation calls for it. It's all about nuance. "Unwelcome" is just too limited.

Reply:
This discussion seems to be more dense that it really needs to be. First, at your point to me makes sense if was a discussion direct to Contributor Convenant.

In first place, an open source project isn't a good place (in my opinion) to start some type of interaction with another person, even with the best intentions. Enforce a code of conduct isn't about punishing bad people, is about take the appropriate action. What action? I don't know, depends on the case, maybe?

Maintain a project isn't about only code or devops, and as an open source project, is about maintain also the community, as an open and welcoming environment. Why project maintainers shouln't interpose if needed in case of threat to the community itself?

Reply:
I think the main problem lies in the "unwelcome" part as it is vague about what can and can't be considered "unwelcome". Maybe we should set it against all kind of sexual attention and advances and acting according the situation.

About how we act if needed, I think we must have some kind of guidelines or define a process (doesn't need to be in this document as it will be mostly between the maintainers) how we proceed to be sure that anyone will take more or less the same approach to handle the situation

Reply:
The discussion is dense because the issue is complex. And it is relevant because the proposal is to include the Contributor Convenant into the repository.

I agree with you both. I also think an open source project isn't a good place to start this kind of interaction. And I also think that "unwelcome" is limited and vague. Sexual attention/advances is not defined either. This vagueness is a big problem.

But the main problem about all this isn't that it is vague. The main problem is the mindset that it implies - that how an action is perceived/received and what kind of emotion it elicits in another person is enough to constitute a violation, regardless of the intention. This mindset is reflected in other parts of the document and the FAQ of the Contributor Convenant. And is one kind of mindset that does not contribute to a healthy work environment.

The lack of answers about what kind of action would be appropriate hints to the complexity of the issue, as stated before. And raises doubt about whether project maintainers are the best arbiters for this kind of situation. Remember, this is not any open source project where people volunteer their time and resources freely. This is a project connected to a company's product. Employees of the company interact while contributing to this project, in many ways other than commenting, reviewing and submitting code. Anything that happens during this process, and anything included in this document, could have legal implications. Not to mention impact someone's career and their ability to sustain themselves and their family.